File: LEETCODE/215.cpp
//215. Kth Largest Element in an Array

//Method 1
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
       sort(nums.begin(),nums.end(),greater<int>());
        
       return nums[k-1];
    }
};

//Method 2
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> q; //minheap
        
        int n1=nums.size();
        for(int i=0;i<n1;i++){
            if(i<k)
                q.push(nums[i]);
            else{
                if(nums[i]>q.top()){
                    q.pop();
                    q.push(nums[i]);
                }
            }
        }        
        
        return q.top();
    }
};

File: LEETCODE/LEETCODE-PYTHON/215.py
# 215. Kth Largest Element in an Array
#https://leetcode.com/problems/kth-largest-element-in-an-array/

# Mehod-1 using sorting (TC-0(nlogn))
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        nums.sort(reverse=True)

        return nums[k-1]

# Method-2 using heap (TC-O(nlok))
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        heap=[]

        for num in nums:
            heapq.heappush(heap,num)

            if len(heap)>k:
                heapq.heappop(heap)

        return heap[0]